### PR TITLE
Setup profile API

### PR DESCRIPTION
**Pull request by GiuliaTeggi**
### Ticket:

- [15378](https://dev.azure.com/OGCIO-Digital-Services/Digital%20Services%20Programme/_workitems/edit/15378)

### Description

- Setup profile API app
- Add `/addresses` endpoints to get, create, update and delete addresses
- Update the building blocks SDK
- Update profile app components to use such endpoints

Entitlements and user details endpoints will be added in follow-up prs

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [x] **New feature**
- [x] **Dev change**

### Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works

### Screenshots:

Screenshot 2024-04-17 at 18 51 24

### CODE REVIEW

File: apps/profile-api/app.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  app.register(fastifySwagger, {
+    openapi: {
+      info: {
+        title: "OGCIO Profile API",
+        description: "API for OGCIO Profile Service",
+        version: "0.1.0",
+      },
+      tags: [
+        {
+          name: "Providers",
```
Comment (alfonsograziano) [inline apps/profile-api/app.ts:40]: Do you need this tag?

Comment (GiuliaTeggi) [reply apps/profile-api/app.ts:40]: Nope :D will change it to addresses

File: apps/profile-api/package.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
+{
+  "name": "profile-api",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "node --import=tsx --test ./test/*.ts",
+    "start": "node --import tsx index.ts",
+    "dev": "nodemon | pino-pretty",
+    "generate-client": "openapi-generator-cli generate -i openapi-definition.yml -g typescript-axios -o ../profile/client/autogenerated",
```
Comment (alfonsograziano) [inline apps/profile-api/package.json:10]: This has been an early experiment, but during some API development @nnorbert found a couple of bugs. In payments we're slowly moving to the `openapi-typescript` generator (the one working with the `generate-ts-client` command)

File: apps/profile-api/openapitools.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "$schema": "./node_modules/@openapitools/openapi-generator-cli/config.schema.json",
```
Comment (alfonsograziano) [inline apps/profile-api/openapitools.json:2]: Are you using `openapi-generator`?

**Review by alfonsograziano — APPROVED** [review]
Apart from a couple of really minor comments, LGTM :)

File: apps/profile-api/package.json (hunk 2 above)
Comment (GiuliaTeggi) [reply apps/profile-api/package.json:10]: Ok, when you have this working with openapi-typescript I can update the profile API too to use that

**Comment by edge33** [comment]
LGTM

**Pull request merged**